tests/part_test.py

```diff
@@ -912,6 +912,69 @@ def test_get_part_by_pos(self):
         self.assertEqual(ret.start, ps4.start + ps4.size)
         self.assertLessEqual(ret.size, (100 * 1024**2) - (ps4.start + ps4.size))
 
+class PartCreateResizePartCase(PartTestCase):
+    def test_create_resize_part_simple(self):
+        """Verify that it is possible to create and resize a parition"""
+
+        # we first need a partition table
+        succ = BlockDev.part_create_table (self.loop_dev, BlockDev.PartTableType.MSDOS, True)
+        self.assertTrue(succ)
+
+        # create a maximal partition
+        ps = BlockDev.part_create_part (self.loop_dev, BlockDev.PartTypeReq.NORMAL, 2048*512, 0, BlockDev.PartAlign.OPTIMAL)
+        initial_start = ps.start
+        initial_size = ps.size
+
+        new_size = 20 * 1000**2  # resize to 20 MB
+        succ = BlockDev.part_resize_part (self.loop_dev, ps.path, new_size, BlockDev.PartAlign.OPTIMAL)
+        self.assertTrue(succ)
+        ps = BlockDev.part_get_part_spec(self.loop_dev, ps.path)
+        self.assertEqual(initial_start, ps.start)  # offset must not be moved
+        self.assertGreaterEqual(ps.size, new_size)  # at least the requested size,
+        self.assertLess(ps.size, new_size + 8 * 1024**2)  # but also not too big
```

Comment:
8 MiB is quite a big difference from 20 MiB. AFAIK, we have all the test disks set to have optimal alignment ~1 MiB so there should never be such a big difference.